scrapy/downloadermiddlewares/redirect.py

```diff
@@ -27,6 +27,7 @@ def _build_redirect_request(
     redirect_request = source_request.replace(
         url=url,
         **kwargs,
+        cls=None,
```

Comment:
Without this line the type of `redirect_request` is inferred to be `Any`.

Reply:
Could you please add a test for this case? I guess it's `**kwargs` handling?

Reply:
I think it's because it cannot know that `kwargs` passed to `_build_redirect_request` doesn't contain `cls`.

Reply:
Also, what test do you mean?

Reply:
I've added tests that check that the result is `Any` unless `cls` was also passed.

Reply:
Not sure if it's a useful thing to check :)